data/contract_docs/logs.md

```diff
@@ -0,0 +1,92 @@
+<!-- markdownlint-configure-file {
+  "MD013": false,
+  "MD033": false
+} -->
+
+# Logs
+
+Status: EARLY Draft/Proposal
+
+## LogLines
+
+Version: 0.0
+
+### Properties and field requirements
+
+- **Time field** - _required_
+  - There must be at least one non nullable time field
+  - If there are multiple time fields present, following will decide the priority
+    - First matching time field with name `timestamp`
+- **Message field** - _required_
+  - There must be at lease one non nullable string field must present
+  - If more than one string fields found, the following will decide the priority
+    - First matching string field with name `body`
+- **Severity field** - _optional_
+  - This is optional field
+  - Level/Severity of the log line can be represented with this field.
+  - First matching string field with name `severity` will be considered as severity field
+  - If no level field found, consumers/client will decide the log level. Example: logs panel will try to parse the message field and determine the log level
+  - Log level can be one of the values specified in the docs [here](https://grafana.com/docs/grafana/latest/explore/logs-integration/)
+- **ID field** - _optional_
+  - This optional field
+  - Unique identified of the log line
+  - This have to be a string field. (either nullable or non-nullable string field)
+  - First matching string field with name `id` will be considered as severity field
+  - If no id field found, consumers/client will decide the id field as required.
+- **Attributes field** - _optional_
+  - This is an optional field
+  - This field is also known as labels
+  - This field represent additional attributes of the log line. This is also known as labels field.
+  - Field type must be json raw message type. Example value: `{}`, `{"hello":"world", "foo": 123.45, "bar" :["yellow","red"], "baz" : { "name": "alice" }}`
+    - Should not be empty string.
+    - Value should be represented with `Record<string,any>` type in javascript.
+  - First matching string field with name `attributes` will be considered as attributes field
+- **NanoSecond Time field** - _optional_
+  - When the log line have sub-milli second precisions, regular time field not suitable to represent them.
+  - Min Value: null/0
```

Comment:
question: does allowing `null` helps us with something?

the alternative is to use `0` here.

maybe `null` can mean "i do not know the rest of the timestamp"? not sure. 

in other words: if `null` does not give us anything extra, i'd say we make this a non-nullable-field. of course, if it helps with something, i'm fine with it.

Reply:
https://github.com/grafana/grafana-plugin-sdk-go/pull/647 exists now, so marking this part of the conversation resolved.